server/aws-lsp-codewhisperer/src/language-server/netTransformServer.ts

```diff
@@ -18,7 +25,48 @@ export const NetTransformServerFactory: (
             params: ExecuteCommandParams,
             _token: CancellationToken
         ): Promise<any> => {
-            //Placeholder for logic
+            try {
+                const client = createService(credentialsProvider)
+                const transformHandler = new TransformHandler(client, workspace)
+                switch (params.command) {
+                    case 'aws/qNetTransform/startTransform': {
+                        const userInputrequest = params as QNetStartTransformRequest
+                        logging.log('prepare artifact for solution: ' + userInputrequest.SolutionRootPath)
+                        return await transformHandler.startTransformation(userInputrequest)
+                    }
+                    case 'aws/qNetTransform/getTransform': {
+                        const request = params as QNetGetTransformRequest
+                        logging.log('Calling getTransform request with job Id: ' + request.TransformationJobId)
+                        return await transformHandler.getTransformation(request)
+                    }
+                    case 'aws/qNetTransform/pollTransform': {
```

Comment:
Reading the rest of the CR, but why would the client poll the transform? The server can report progress and partial results when the the server gets status, and that logic is owned by the server. The client doesn't have to worry about that.

Reply:
When you are saying client does that mean Toolkit and server mean Language server?

We do need to poll the status of the job do display different status on the UI

Reply:
how this partial results work? is there any example how client get notification/result from server side?

Reply:
One important note is that custom commands do not have support for partial results but they do have support for work done progress reporting. 

Here (https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_executeCommand) you can see that `interface ExecuteCommandParams extends WorkDoneProgressParams`. `WorkDoneProgressParams` includes a `workDoneToken` which is a token of type string or number. If a client generates and provides a workDoneToken(usually achieved by creating a random uuid) when making a request to the server, the server can report progress on this token to client. This mechanism can be used instead of registering a new custom command for polling the transform status

- LSP documentation for client initiated work done progress: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#clientInitiatedProgress
- LSP documentation for partial results: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#partialResults

Reply:
There're two polling requests in our workflow. it seems that these polling requests may not be applicable to all of our polling scenarios. Here's our current workflow:

1. once receiving an accepted response from startTransformRequest, the client initiates a call to getJobStatus to poll the status.

2. When the jobStatus retrieved from the GET API transitions from "accepted" to "transforming," the backend service generates a transformPlan.

3. the client begins polling another API called getTransformPlan to update the plan object locally.Note: the plan is not static; it undergoes dynamic changes in both status and plan steps. Each time the plan changes, the updated plan needs to be propagated to the client.

4. the client continues polling getJobStatus to monitor the job status until it's marked as completed.

I think we can use the work done progress reporting as optimization to replace the getJobStatus polling. we have to make change on both the client and server sides. Given that we currently have a different set of PRs under review on the client side, can we take this effort in a new PR (both client and server) after closing the current one? We've already documented this proposed change in our ticket for reference.